rusk/src/lib/pow_verifier.rs

```diff
@@ -0,0 +1,76 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use blake2::{Blake2s256, Digest};
+
+pub const POW_DIFFICULTY: usize = 16;
+
+#[derive(Debug)]
+pub struct PoW;
+
+impl PoW {
+    /// Produces a nonce such that hash of a given collection of bytes
+    /// concatenated with the nonce has difficulty `difficulty`
+    /// (meaning, its last `difficulty` bits will be zero)
```

Comment:
I believe trailing bits are usually considered the first bits, not the last (hence they are "trailing" or "leading")